sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -195,16 +195,21 @@ case class AlterTableRenameCommand(
       DDLUtils.verifyAlterTableType(catalog, table, isView)
       // If an exception is thrown here we can just assume the table is uncached;
       // this can happen with Hive tables when the underlying catalog is in-memory.
-      val wasCached = Try(sparkSession.catalog.isCached(oldName.unquotedString)).getOrElse(false)
-      if (wasCached) {
+      // If `optStorageLevel` is defined, the old table was cached.
+      val optStorageLevel = Try {
+        val optCachedData = sparkSession.sharedState.cacheManager.lookupCachedData(
+          sparkSession.table(oldName.unquotedString))
+        optCachedData.map(_.cachedRepresentation.cacheBuilder.storageLevel)
+      }.getOrElse(None)
+      optStorageLevel.foreach { _ =>
```

Comment:
nit: `if (optStorageLevel.isDefine)`. It's clearer to check if the table is cached or not.

Reply:
+1

Reply:
updated.